plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -139,11 +140,27 @@ fu_thunderbolt_device_get_version(FuThunderboltDevice *self, GError **error)
 			    version_raw);
 		return FALSE;
 	}
-	if (!fu_strtoull(split[0], &version_major, 0, G_MAXUINT64, FU_INTEGER_BASE_16, error))
+	if (!fu_strtoull(split[0], &version_major, 0, G_MAXUINT64, FU_INTEGER_BASE_10, error))
 		return FALSE;
-	if (!fu_strtoull(split[1], &version_minor, 0, G_MAXUINT64, FU_INTEGER_BASE_16, error))
-		return FALSE;
-	version = g_strdup_printf("%02x.%02x", (guint)version_major, (guint)version_minor);
+
+	/* thunderbolt versioning convention on controller type only, don't apply retimer */
+	if (FU_IS_THUNDERBOLT_CONTROLLER(self)) {
+		guint64 minor_raw = 0;
+		if (!fu_strtoull(split[1], &minor_raw, 0, G_MAXUINT64, FU_INTEGER_BASE_16, error))
+			return FALSE;
+		version_minor = ((minor_raw & 0x3f) >> 4) * 10 + (minor_raw & 0x0f);
+	} else {
+		if (!fu_strtoull(split[1],
+				 &version_minor,
+				 0,
+				 G_MAXUINT64,
+				 FU_INTEGER_BASE_10,
+				 error))
+			return FALSE;
+	}
+
+	version = g_strdup_printf("%u.%u", (guint)version_major, (guint)(version_minor));
+	g_debug("Thunderbolt version_raw: %s, convention: %s", g_strchomp(version_raw), version);
 	fu_device_set_version(FU_DEVICE(self), version);
```

Comment:
Considering it only applies to the controller; I wonder if perhaps there should be a separate controller and retimer vfunc instead of keeping it all in `fu_thunderbolt_device_get_version`?

@hughsie, thoughts?